pdf/pdfcanvas.cpp

```diff
@@ -469,55 +501,61 @@ QSGNode* PDFCanvas::updatePaintNode(QSGNode *node, QQuickItem::UpdatePaintNodeDa
             }
         }
 
-        if (page.texture && showPage) {
-            if (page.rectPlaceholder) {
-                delete t->firstChild();
-                t->removeAllChildNodes();
-            }
-            QSGSimpleTextureNode *tn = static_cast<QSGSimpleTextureNode *>(t->firstChild());
-            if (!tn) {
-                tn = new QSGSimpleTextureNode;
-                t->appendChildNode(tn);
+        if (showPage) {
+            // Node hierachy:
+            // t
+            // |-bg
+            // |  |-tn
+            // |  |-n
+            // |  | |- link1
+            // |  | |- link2...
+            QSGSimpleRectNode *bg = static_cast<QSGSimpleRectNode*>(t->firstChild());
+            if (!bg) {
+                bg = new QSGSimpleRectNode;
+                t->appendChildNode(bg);
             }
+            bg->setRect(0., 0., page.rect.width(), page.rect.height());
+            bg->setColor(d->pagePlaceholderColor);
```

Comment:
I agree, I've moved it some lines above, when bg object is created. I'm just wondering when ambiance is changed if there can be a case when the page place holder is visible and thus not in the right colour. The only case I imagine is ambiance changed by a timer and the page place holder is visible. But this is very rare situation…